src/PixiEditor.ChangeableDocument/Changes/Selection/SelectEllipse_UpdateableChange.cs

```diff
@@ -26,6 +27,8 @@ public void Update(RectI borders)
     public override bool InitializeAndValidate(Document target)
     {
         originalPath = new VectorPath(target.Selection.SelectionPath);
+        documentConstraint = new VectorPath();
+        documentConstraint.AddRect(new RectI(VecI.Zero, target.Size));
```

Comment:
You can easily omit this constraint by resizing canvas, I doubt it's a problem though. It applies to all constraints you made